### PR TITLE
Fix `-Wsign-compare` warnings

### PR DESCRIPTION
**Pull request by greg7mdp**

### CODE REVIEW

File: libraries/chain/CMakeLists.txt [hunk 1]
```diff
@@ -36,6 +36,7 @@ if("eos-vm-oc" IN_LIST EOSIO_WASM_RUNTIMES)
 
    set_source_files_properties(webassembly/runtimes/eos-vm-oc/LLVMJIT.cpp PROPERTIES COMPILE_FLAGS "--std=gnu++17")
    set_source_files_properties(webassembly/runtimes/eos-vm-oc/LLVMEmitIR.cpp PROPERTIES COMPILE_FLAGS "--std=gnu++17")
+   set_source_files_properties(webassembly/runtimes/eos-vm-oc/switch_stack_linux.s PROPERTIES COMPILE_FLAGS "-Wno-unused-command-line-argument")
```
Comment (spoonincode) [inline libraries/chain/CMakeLists.txt:39]: I think this is a fine fix.. but I wonder why cmake is operating this way :thinking:  it clearly knows that `.s` are assembly files -- not c/c++ files -- so why is it tacking on all the c/c++ options like `-O3 -DNDEBUG` etc

File: plugins/state_history_plugin/tests/session_test.cpp [hunk 2]
```diff
@@ -530,7 +530,7 @@ BOOST_FIXTURE_TEST_CASE(test_session_fork, state_history_test_fixture) {
       std::vector<eosio::state_history::block_position> have_positions;
       eosio::state_history::state_result result;
       // we should get 4 consecutive block result
-      for (int i = 0; i < 4; ++i) {
+      for (uint32_t i = 0; i < 4; ++i) {
```
Comment (heifner) [inline plugins/state_history_plugin/tests/session_test.cpp:533]: `size_t` is likely better here.

File: plugins/state_history_plugin/tests/session_test.cpp [hunk 3]
```diff
@@ -580,7 +580,7 @@ BOOST_FIXTURE_TEST_CASE(test_session_fork, state_history_test_fixture) {
 
       eosio::state_history::state_result fork_result;
       // we should now get data for fork 3,4
-      for (int i = 2; i < 4; ++i) {
+      for (uint32_t i = 2; i < 4; ++i) {
```
Comment (heifner) [inline plugins/state_history_plugin/tests/session_test.cpp:583]: `size_t` is likely better here.

**Review by heifner — APPROVED** [review]

File: libraries/chain/CMakeLists.txt (hunk 1 above)
Comment (greg7mdp) [reply libraries/chain/CMakeLists.txt:39]: beats me!

File: plugins/state_history_plugin/tests/session_test.cpp (hunk 2 above)
Comment (greg7mdp) [reply plugins/state_history_plugin/tests/session_test.cpp:533]: Hum, I'm not sure. The warning was when comparing with `r.this_block->block_num` which is a `uint32_t`.
If it was a comparison with `<container>.size()` then I would have used `size_t`.

File: plugins/state_history_plugin/tests/session_test.cpp (hunk 3 above)
Comment (greg7mdp) [reply plugins/state_history_plugin/tests/session_test.cpp:583]: same as above.

File: plugins/state_history_plugin/tests/session_test.cpp (hunk 2 above)
Comment (heifner) [reply plugins/state_history_plugin/tests/session_test.cpp:533]: Ah, I thought it was likely complaining about the `written_data[i]`. `uint32_t` then.

**Review by linh2931 — APPROVED** [review]
Thanks for those large amount of edits.

**Pull request merged**